src/org/mozilla/javascript/AbstractEcmaObjectOperations.java

```diff
@@ -0,0 +1,111 @@
+package org.mozilla.javascript;
+
+/**
+ * Abstract Object Operations as defined by EcmaScript
+ *
+ * @see <a href="https://262.ecma-international.org/11.0/#sec-operations-on-objects">Abstract
+ *     Operations - Operations on Objects</a>
+ */
+class AbstractEcmaObjectOperations {
+    static enum INTEGRITY_LEVEL {
+        FROZEN,
+        SEALED
+    }
+
+    /**
+     * Implementation of Abstract Object operation testIntegrityLevel as defined by EcmaScript
+     *
+     * @param o
+     * @param level
+     * @return boolean
+     * @see <a
+     *     href="https://262.ecma-international.org/11.0/#sec-testintegritylevel">TestIntegrityLevel</a>
+     */
+    static boolean testIntegrityLevel(Object o, INTEGRITY_LEVEL level) {
+        ScriptableObject obj = ScriptableObject.ensureScriptableObject(o);
+        Context cx = Context.getCurrentContext();
+
+        if (obj.isExtensible()) return Boolean.FALSE;
```

Comment:
Is there a reason not to just return "true" and "false" from this function, since it returns a boolean anyway?

Reply:
That was there already in the old code, but didn't look like it's needed. Will fix